packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -722,8 +772,17 @@ export class LocalFluidDataStoreContextBase extends FluidDataStoreContext {
         assert(this.isRootDataStore !== undefined, "isRootDataStore should be available in local data store context");
         return {
             pkg: this.pkg,
-            isRootDataStore: this.isRootDataStore,
             snapshot: this.snapshotTree,
+            isRootDataStore: this.isRootDataStore,
+        };
+    }
+
+    protected async getInitialGCDetails(): Promise<IGCDetails> {
+        assert(this.isRootDataStore !== undefined, "isRootDataStore should be available in local data store context");
+        return {
+            isRootNode: this.isRootDataStore,
+            // Local data stores don't have initial GC nodes. These will be generated when it is summarized.
+            gcNodes: [],
```

Comment:
Similar concern as above - using gcNodes = [] anywhere is incorrect (unless I missed something).
It feels like this function should not be called for local data store before first summary, and actual data should be populated and returned after any successful summary

Reply:
Updated as per my comment above.